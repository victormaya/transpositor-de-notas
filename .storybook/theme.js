import { create } from '@storybook/theming'
import theme from '../src/styles/theme'
import logo from '../public/img/nextjs-logo.svg'

export default create({
  base: 'dark',
  colorSecondary: theme.colors.primary,

  brandTitle: 'Boilerplate',
  brandUrl: 'https://alquipo.dev/',
  brandImage: logo
})
