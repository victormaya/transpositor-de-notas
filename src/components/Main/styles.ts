import styled from 'styled-components'

export const Wrapper = styled.main`
  background-color: ${(props) => props.theme.colors.mainBg};
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`

export const Logo = styled.img`
  color: white;
  width: 25rem;
  margin-bottom: 2rem;
`

export const Title = styled.h1`
  font-size: 2.5rem;
`

export const Description = styled.h2`
  font-size: 2rem;
  font-weight: 400;
`

export const Illustration = styled.div`
  margin-top: 3rem;
  width: min(30rem, 100%);
  height: min(30rem, 100%);
  position: relative;
`
export const Button = styled.button`
  border: none;
  background-color: ${(props) => props.theme.colors.secondary};
  width: 12rem;
  height: 3rem;
  cursor: pointer;
  border-radius: 0.5rem;
  margin-top: 1rem;

  font-weight: bold;
`
