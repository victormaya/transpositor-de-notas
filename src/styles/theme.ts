export default {
  grid: {
    container: '1060px'
  },
  border: {
    radius: '0.4rem'
  },
  font: {
    family: "'Poppins', sans-serif",
    light: 300,
    normal: 400,
    bold: 700,
    sizes: {
      small: '1.2rem',
      medium: '1.5rem',
      large: '2.0rem',
      xlarge: '2.4rem',
      xxlarge: '3.6rem'
    }
  },
  colors: {
    green: '#41CBBF',
    dark: '#164E63',
    light: '#F8FAFC',
    gray: '#BEC9DF'
  },
  spacings: {
    small: '1rem',
    medium: '1.5rem',
    large: '3rem',
    xlarge: '6rem',
    xxlarge: '9rem'
  }
} as const
