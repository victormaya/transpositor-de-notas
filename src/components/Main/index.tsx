import Image from 'next/image'
import { useRouter } from 'next/router'

import * as S from './styles'

export type MainProps = {
  title?: string
  description?: string
  page?: string
}
const Main = ({
  title = 'Boilerplate NextJS - Alquipo Neto',
  description = 'TypeScript, ReactJS, NextJS e Styled Components',
  page
}: MainProps) => {
  const router = useRouter()

  return (
    <S.Wrapper>
      <S.Logo src="/img/nextjs-logo.svg" alt="Imagem escrito NEXT.JS" />

      <S.Title>{title}</S.Title>
      <S.Description>{description}</S.Description>
      <S.Illustration>
        <Image
          src="/img/hero-illustration.svg"
          alt="Um desenvolvedor de frente para uma tela com código."
          layout="fill"
          objectFit="contain"
        />
      </S.Illustration>
      <S.Button onClick={() => router.push(page)}>Mudar de Página</S.Button>
    </S.Wrapper>
  )
}

export default Main
