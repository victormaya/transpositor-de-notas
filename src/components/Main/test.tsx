import theme from 'styles/theme'
import { render, screen, fireEvent } from 'utils/test-utils'

import Main from '.'

// eslint-disable-next-line @typescript-eslint/no-var-requires
const useRouter = jest.spyOn(require('next/router'), 'useRouter')
const push = jest.fn()

useRouter.mockImplementation(() => ({
  push
}))

describe('<Main/>', () => {
  it('should render the heading', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', {
        name: /Boilerplate NextJS - Alquipo Neto/i
      })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyle({
      'background-color': theme.colors.mainBg
    })
  })

  it('should handle the button switch page', () => {
    render(<Main page="/page2" />)

    fireEvent.click(screen.getByRole('button', { name: /Mudar de PÃ¡gina/i }))

    expect(push).toHaveBeenCalledWith('/page2')
  })
})
