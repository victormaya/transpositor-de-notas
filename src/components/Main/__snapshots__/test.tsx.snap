// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main/> should render the heading 1`] = `
.c0 {
  background-color: #06092B;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  color: white;
  width: 25rem;
  margin-bottom: 2rem;
}

.c2 {
  font-size: 2.5rem;
}

.c3 {
  font-size: 2rem;
  font-weight: 400;
}

.c4 {
  margin-top: 3rem;
  width: min(30rem,100%);
  height: min(30rem,100%);
  position: relative;
}

.c5 {
  border: none;
  background-color: #3CD3C1;
  width: 12rem;
  height: 3rem;
  cursor: pointer;
  border-radius: 0.5rem;
  margin-top: 1rem;
  font-weight: bold;
}

<main
  class="c0"
>
  <img
    alt="Imagem escrito NEXT.JS"
    class="c1"
    src="/img/nextjs-logo.svg"
  />
  <h1
    class="c2"
  >
    Boilerplate NextJS - Alquipo Neto
  </h1>
  <h2
    class="c3"
  >
    TypeScript, ReactJS, NextJS e Styled Components
  </h2>
  <div
    class="c4"
  >
    <span
      style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
    >
      <img
        alt="Um desenvolvedor de frente para uma tela com código."
        data-nimg="fill"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
      />
      <noscript />
    </span>
  </div>
  <button
    class="c5"
  >
    Mudar de Página
  </button>
</main>
`
